graphql/pupil/fields.ts

```diff
@@ -110,4 +110,18 @@ export class ExtendFieldsPupilResolver {
     async canJoinSubcourses(@Root() pupil: Required<Pupil>) {
         return canJoinSubcourses(pupil);
     }
-}
\ No newline at end of file
+
+    @FieldResolver(returns => [Boolean])
+    @Authorized(Role.ADMIN)
+    async isParticipant(@Root() pupil: Required<Pupil>, @Arg("subcourseId") subcourseId: number) {
```

Comment:
I'd prefer if the resolver would be on the Subcourse, and would by default take the session pupil (and support the pupilId as an optional parameter)

Reply:
ahh ok that was kind of my first approach but then I saw the isParticipant field in pupil and thought that a fieldResolver should resolve that field haha, because in subcourse is no such field

Reply:
Yeah pupil.isParticipant means the user can generally participate in courses and pupil.isPupil means the pupil can take one-to-one tutoring ... Maybe we should somewhen rename those ...

Reply:
Ok I see, but what field in subcourses should be resolved then? Or should it be a query

Reply:
No, a field resolver, subcourse.isParticipant(subcourseId?)

Reply:
Ok then maybe I am misunderstanding something because I thought a field resolver would resolve a field in the schema so making a field resolver in subcourse would require a field "isParticipant" in the subcourse entity. But there is no such field. I will take a look at it again tomorrow

Reply:
Ok, but there also is no field in the Graphql entity and I get the following error when trying to resolve that field with this resolver:
![image](https://user-images.githubusercontent.com/48623649/189083257-f5abc5b1-e235-4295-a5a7-65152d189273.png)
Error:
![image](https://user-images.githubusercontent.com/48623649/189082789-1915fe75-45fa-4308-8223-b2022a9cfa5b.png)

So should I create a new field or am I still getting something wrong.

Reply:
@Return(_type => Boolean) ...

Reply:
ahhh damn I didnt see that the entire time. Dont know why I added brackets there. Maybe something left over from copy pasting